src/diagnostics/radiation_diagnostics.jl

```diff
@@ -57,24 +75,25 @@ function compute_rsdt!(
     radiation_mode::T,
 ) where {T <: RRTMGPI.AbstractRRTMGPMode}
     nlevels = Spaces.nlevels(axes(state.c))
+    z_max = Spaces.z_max(axes(state.f))
+    planet_radius = CAP.planet_radius(cache.params)
+    FT = eltype(cache.params)
+
     if isnothing(out)
-        return copy(
-            Fields.level(
-                Fields.array2field(
-                    cache.radiation.rrtmgp_model.face_sw_flux_dn,
-                    axes(state.f),
-                ),
-                nlevels + half,
-            ),
-        )
-    else
-        out .= Fields.level(
+        out = copy(
```

Comment:
I think you accidentally removed Fields.level(..., nlevels+half) here and in other TOA diagnostics?